src/Hamiltonians/abstract.jl

```diff
@@ -12,69 +12,59 @@ end
 BitStringAddresses.num_modes(h::AbstractHamiltonian) = num_modes(starting_address(h))
 
 """
-    logbinomialapprox(n, k)
+    dimension(h::AbstractHamiltonian, addr=starting_address(h))
+    dimension(addr::AbstractFockAddress)
 
-Approximate formula for log of binomial coefficient. [Source](https://en.wikipedia.org/wiki/Binomial_coefficient#Bounds_and_asymptotic_formulas)
-"""
-logbinomialapprox(n,k) = (n+0.5)*log((n+0.5)/(n-k+0.5))+k*log((n-k+0.5)/k) - 0.5*log(2π*k)
-
-"""
-    dimension([::Type{T}], h)
+Return the estimated dimension of Hilbert space.
```

Comment:
```suggestion
Return the estimated dimension of Hilbert space as a `BigInt`.
```
I'd make this more explicit just in case someone runs into issues with expecting an `Int` and getting a `BigInt`.

Reply:
Yeah, I didn't want to commit to `BigInt`. Say if (in the future) we invent an address type that has an infinite address space. E.g. a Fock space where the number of particles is not fixed. The you could return `Inf`, which is a `Float64` and larger than all `BigInt`s. I'm not aware that `BigInt` has its own infinite type instance.

Is there a reason why it would be beneficial to commit to returning a specific number type? Wouldn't that also break AD with dual numbers, etc?

Reply:
Ah, I didn't necessarily want us to commit to it. I meant to include it as a warning because a user may expect it to be an `Int` (julia prints `BigInt`s and `Int`s in the same way).

Reply:
OK, I added a warning.